mkosi/manifest.py

```diff
@@ -197,6 +199,31 @@ def record_deb_packages(self, root: Path) -> None:
 
             source_package.add(package)
 
+    def record_pkg_packages(self, root: Path) -> None:
+        package_descriptions = [str(p) for p in (root / "/var/lib/pacman/local").glob("*/desc")]
+        awkprog = """\
+BEGIN { RS="\\n\\n"; FS="\\n"; }
+/%NAME%/ { name=$2 };
+/%VERSION%/ { version=$2 };
+/%BASE%/ { base=$2 };
+/%ARCH%/ { arch=$2 };
+ENDFILE { print name, version, base, arch }
+"""
+        c = run(["awk", awkprog] + package_descriptions, stdout=PIPE, text=True)
```

Comment:
hmpf, you shell out to awk for a simple parser like that? Let's maybe not do that... The awk expression above sounds simple enough to just implement in python.

Reply:
Fixed :)